.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,52 @@
+name: Build and test
+
+on:
+  pull_request:
+    types:
+      - opened
+      - ready_for_review
+      - synchronize
+  push:
+    branches:
+      - master
+
+permissions:
+  contents: read
+
+jobs:
```

Comment:
Separated `rake spec` into 2jobs (3 tasks) to visualize the results more friendly.